### PR TITLE
Feature: Add i18n

### PR DESCRIPTION
**Pull request by juanchoperezj**
### Pull request type

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

---

### Description

Support multiple languages and RTL using `i18next` and `react-i18next`.
Strings used to display the translations have Typescript autocompletion.

---

#### Jira board reference

- 

---

#### Notes:

- react-native-restart package was added to restart the app after a language change
- Device language detection wasn't implemented, which would bias the base to use a specific package.

---

#### Tasks:

- [ ] Tested on IOS
- [ ] Tested on Android
- [ ] Tested on a small device
- [ ] Tested on a real device
- [ ] Tested all flows related with this PR changes
- [ ] Tested accessibility
- [ ] Added tests

---

#### Preview

[i18n feature.webm](https://github.com/rootstrap/react-native-base/assets/17632690/e0c078f6-fc54-4c27-ba7e-675fcb748a11)

https://github.com/rootstrap/react-native-base/assets/17632690/d5d7f228-417e-4767-bea1-1b4849b81047

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ module.exports = {
     '^features/(.*)$',
     '^network/(.*)$',
     '^storage/(.*)$',
+    '^common/(.*)$',
+    '^localization/(.*)$',
```
Comment (jpirazusta) [inline .prettierrc.js:20]: Maybe sort these new ones alphabetically also?

Comment (juanchoperezj) [reply .prettierrc.js:20]: Good catch!

File: src/localization/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+export const useLanguage = () => {
+  const currentLanguage = Stores.GlobalStorageInstance.get('language');
+
+  const setLanguage = (language: Language) => {
+    i18n.changeLanguage(language, () => {
+      const isRTL = language === 'ar';
+      I18nManager.forceRTL(isRTL);
+      I18nManager.allowRTL(isRTL);
+      Stores.GlobalStorageInstance.set('language', language);
+      Restart.restart();
```
Comment (ManuelJrez) [inline src/localization/hooks.ts:25]: is this fully necessary?? i'm just asking 😅

**Review by ManuelJrez — APPROVED** [review]

File: src/localization/hooks.ts (hunk 2 above)
Comment (juanchoperezj) [reply src/localization/hooks.ts:25]: Which line exactly 😆 ?

**Review by jpirazusta — APPROVED** [review]

**Review by jpirazusta — APPROVED** [review]

**Review by ManuelJrez — APPROVED** [review]

**Pull request merged**